src/GitHub.Api/Application/ApiClient.cs

```diff
@@ -174,16 +182,15 @@ public ApiClient(UriString hostUrl, IKeychain keychain, IGitHubClient githubClie
             return result.Code == LoginResultCodes.Success;
         }
 
-        private async Task CreateRepositoryInternal(NewRepository newRepository, Action<Octokit.Repository, Exception> callback, string organization)
+        private async Task<Octokit.Repository> CreateRepositoryInternal(NewRepository newRepository, string organization)
         {
             try
             {
                 logger.Trace("Creating repository");
 
                 if (!await EnsureKeychainLoaded())
                 {
-                    callback(null, new Exception("Keychain Not Loaded"));
```

Comment:
I mean, the user shouldn't reach here without having their Keychain loaded.
So i guess this code being here is more of a formality...

It's something I hope to clean up as we..
1. Get proper tasks around the API
2. Figure out a clean way to handle authentication errors from the API and Git